examples/nas/search_space_zoo/nasbench201.py

```diff
@@ -0,0 +1,207 @@
+            output = model(data)
+            test_loss += criterion(output, target).item()
+            pred = output.argmax(dim=1, keepdim=True)
+            correct += pred.eq(target.view_as(pred)).sum().item()
+        test_loss /= len(valid_dataloader.dataset)
+        accuracy = 100. * correct / len(valid_dataloader.dataset)
+        logger.info('\nTest set: Average loss: {:.4f}, Accuracy: {}/{} ({:.0f}%)\n'.format(test_loss, correct,
+                                                                                           len(valid_dataloader.dataset), accuracy))
+        model.train()
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser("nb201")
+    parser.add_argument('--stem_out_channels', default=16, type=int)
+    parser.add_argument('--unrolled', default=False, action='store_true')
+    parser.add_argument('--batch_size', default=64, type=int)
+    parser.add_argument('--epochs', default=50, type=int)
+    parser.add_argument('--num_modules_per_stack', default=5, type=int)
+    parser.add_argument('--log-frequency', default=10, type=int)
+    parser.add_argument('--bn_momentum', default=0.1, type=int)
+    parser.add_argument('--bn_affine', default=True, type=bool)
+    parser.add_argument('--bn_track_running_stats', default=True, type=bool)
+    parser.add_argument('--arch', default=None, help='json file which should meet requirements in NAS-Bench-201')
+    parser.add_argument('--visualization', default=False, action='store_true')
+    args = parser.parse_args()
+
+    dataset_train, dataset_valid = datasets.get_dataset("cifar10")
+    model = NASBench201Network(stem_out_channels=args.stem_out_channels,
+                               num_modules_per_stack=args.num_modules_per_stack,
+                               bn_affine=args.bn_affine,
+                               bn_momentum=args.bn_momentum,
+                               bn_track_running_stats=args.bn_track_running_stats)
+
+    optim = torch.optim.SGD(model.parameters(), 0.025)
+    lr_scheduler = torch.optim.lr_scheduler.CosineAnnealingLR(optim, args.epochs, eta_min=0.001)
+    criterion = nn.CrossEntropyLoss()
+
+    if args.arch is not None:
+        logger.info('model retraining...')
+        with open(args.arch, 'r') as f:
+            arch = json.load(f)
+        for trial in query_nb201_trial_stats(arch, 200, 'cifar100'):
+            pprint.pprint(trial)
+        apply_fixed_architecture(model, args.arch)
+        dataloader_train = DataLoader(dataset_train, batch_size=args.batch_size, shuffle=True, num_workers=0)
+        dataloader_valid = DataLoader(dataset_valid, batch_size=args.batch_size, shuffle=True, num_workers=0)
+        train(args, model, dataloader_train, dataloader_valid, criterion, optim,
+              torch.device('cuda' if torch.cuda.is_available() else 'cpu'))
+        exit(0)
+
+    mutator = enas.EnasMutator(model, tanh_constant=1.1, cell_exit_extra_step=True)
```

Comment:
mutator is not necessary. Use the default might be enough.

Reply:
Sure, the code works well without specifying mutator.